src/main/java/mods/railcraft/common/util/inventory/InvTools.java

```diff
@@ -78,6 +78,34 @@ public static ItemStack emptyStack() {
         return null;
     }
 
+    /**
+     * This method changes the original stack!
+     */
+    @Nullable
+    public static ItemStack decSize(@Nullable ItemStack stack) {
```

Comment:
However in the case of the Smoker, it should definitely use depleteItem() for the container support.